### ISSUE TITLE
Support for LastInsertId

### ISSUE THREAD

**Issue by fhmax** [sijms/go-ora#102]
Hi,

I am sending this sql and LastInsertId is alway 0.

prepare, _ := conn.Prepare("declare id number; begin INSERT INTO LOTE (CLIENT_ID, RECEBIDO_EM, QTDE_REGISTROS, STATUS) VALUES(:1, :2, :3, :4) returning numero into id; end;")
result, _ := prepare.Exec("cliente", time.Now(), 1, "PENDENTE")
fmt.Println(result.LastInsertId()) /// always 0 here

How can I get this value?

Thanks

**Comment by sijms** [comment]
it will not work like this
you need to specify output parameter
see connection.go line# 91
`func (conn *Connection) GetNLS() (*NLSData, error) `

you will find an example of output parameters

**Comment by fhmax** [comment]
i'm trying this way
	
```
stmt := go_ora.NewStmt("begin INSERT INTO LOTE (CLIENT_ID, RECEBIDO_EM, QTDE_REGISTROS, STATUS) VALUES(:client, :recebido_em, :qtd, :status) RETURNING NUMERO INTO :id; commit; end;", conn)
	stmt.AddParam("client", "cliente", 40, go_ora.Input)
	stmt.AddParam("recebido_em", time.Now(), 40, go_ora.Input)
	stmt.AddParam("qtd", 1, 40, go_ora.Input)
	stmt.AddParam("status", "PENDENTE", 40, go_ora.Input)
	stmt.AddParam("id", "", 40, go_ora.Output)
```

and i got this error: **TTC error: received code 3 during stmt reading**

**Comment by sijms** [comment]
if id is an integer put 0 instead of ""
`stmt.AddParam("id", 0, 40, go_ora.Output)`

**Comment by fhmax** [comment]
same error with your suggestions.

```
stmt := go_ora.NewStmt("begin INSERT INTO LOTE (CLIENT_ID, RECEBIDO_EM, QTDE_REGISTROS, STATUS) VALUES(:client, :recebido_em, :qtd, :status) RETURNING NUMERO INTO :id; end;", conn)
	stmt.AddParam("client", "cliente", 40, go_ora.Input)
	stmt.AddParam("recebido_em", time.Now(), 40, go_ora.Input)
	stmt.AddParam("qtd", 1, 40, go_ora.Input)
	stmt.AddParam("status", "PENDENTE", 40, go_ora.Input)
	stmt.AddParam("id", 0, 40, go_ora.Output)
```
error: **TTC error: received code 3 during stmt reading**

**Comment by sijms** [comment]
ok I wil investigate the issue

**Comment by sijms** [comment]
I make a commit that fix the problem for you code
```
begin 
    INSERT INTO LOTE (CLIENT_ID, RECEBIDO_EM, QTDE_REGISTROS, STATUS) VALUES(:1, :2, :3, :4) returning numero into id; 
    commit; // important to save data
end;
```
but still the following code will give error
```
INSERT INTO LOTE (CLIENT_ID, RECEBIDO_EM, QTDE_REGISTROS, STATUS) VALUES(:1, :2, :3, :4) returning numero into id
```
or this
```
INSERT INTO LOTE (CLIENT_ID, RECEBIDO_EM, QTDE_REGISTROS, STATUS) VALUES(:1, :2, :3, :4) returning numero
```
note: without using pl/sql

I will fix after finish add tcps feature to the client

**Comment by fhmax** [comment]
ok, the problem has been fixed in version 2.2.1

thanks  a lot

**Comment by sijms** [comment]
now you can run those without errors
```
begin 
    INSERT INTO LOTE (CLIENT_ID, RECEBIDO_EM, QTDE_REGISTROS, STATUS) VALUES(:1, :2, :3, :4) returning numero into :id; 
    -- commit; // no need for commit as the statement become auto commit
end;
```
```
INSERT INTO LOTE (CLIENT_ID, RECEBIDO_EM, QTDE_REGISTROS, STATUS) VALUES(:1, :2, :3, :4) returning numero into :id
```

the last statement
```
INSERT INTO LOTE (CLIENT_ID, RECEBIDO_EM, QTDE_REGISTROS, STATUS) VALUES(:1, :2, :3, :4) returning numero
```
 is not supported by oracle
